src/Domain/Model/PayPalPayment.php

```diff
@@ -30,47 +28,70 @@ public function __construct( int $id, Euro $amount, PaymentInterval $interval )
 		$this->bookingData = [];
 	}
 
-	public function hasExternalProvider(): bool {
-		return true;
-	}
-
 	public function getValuationDate(): ?DateTimeImmutable {
 		return $this->valuationDate;
 	}
 
-	public function isCompleted(): bool {
+	public function isBooked(): bool {
 		return $this->valuationDate !== null && !empty( $this->bookingData );
 	}
 
+	public function canBeBooked( array $transactionData ): bool {
+		// Unbooked payments can always be booked
+		if ( !$this->isBooked() ) {
+			return true;
+		}
+		// Booked "parent" payments (where "parentPayment" is null) can be booked as followup payments
+		return $this->parentPayment === null;
+	}
+
 	/**
-	 * @param array<string,mixed> $transactionData
-	 *
-	 * @return void
+	 * @param array<string,mixed> $transactionData Payment information from PayPal
+	 * @param PaymentIDRepository $idGenerator Used for creating followup payments
+	 * @return PayPalPayment
 	 *
-	 * @throws DomainException
 	 */
-	public function bookPayment( array $transactionData ): void {
+	public function bookPayment( array $transactionData, PaymentIDRepository $idGenerator ): PayPalPayment {
 		$transformer = new PayPalBookingTransformer( $transactionData );
-		if ( $this->isCompleted() ) {
+		if ( !$this->canBeBooked( $transactionData ) ) {
 			throw new DomainException( 'Payment is already completed' );
 		}
+
+		if ( $this->isFollowupBooking( $transactionData ) ) {
+			return $this->createFollowUpPayment( $transactionData, $idGenerator );
+		}
+
 		$this->bookingData = $transformer->getBookingData();
 		$this->valuationDate = $transformer->getValuationDate();
+		return $this;
 	}
 
 	/**
 	 * @return array<string,mixed>
 	 */
 	public function getLegacyData(): array {
+		// TODO checked for booked status
 		return ( new PayPalBookingTransformer( $this->bookingData ) )->getLegacyData();
 	}
 
-	public function createFollowUpPayment( int $followUpPaymentId ): Payment {
-		if ( $this->parentPayment !== null ) {
-			throw new \RuntimeException( 'You can only create follow-up payments from initial, non-follow-ip payments' );
-		}
-		$payment = new PayPalPayment( $followUpPaymentId, $this->amount, $this->interval );
-		$payment->parentPayment = $this;
-		return $payment;
+	/**
+	 * Create a booked followup payment
+	 *
+	 * @param array<string,mixed> $transactionData
+	 * @param PaymentIDRepository $idGenerator
+	 * @return PayPalPayment
+	 */
+	private function createFollowUpPayment( array $transactionData, PaymentIDRepository $idGenerator ): PayPalPayment {
+		$followupPayment = new PayPalPayment( $idGenerator->getNewID(), $this->amount, $this->interval );
+		$followupPayment->parentPayment = $this;
+		return $followupPayment->bookPayment( $transactionData, $idGenerator );
+	}
+
+	/**
+	 * @param array<string,mixed> $transactionData
+	 * @return bool
+	 */
+	private function isFollowupBooking( array $transactionData ): bool {
```

Comment:
yeah, you caught me here violating YAGNI. I thought it would be handy to check for double vs followup booking (comparing `txn_id` in the booking data). Since we don't do that at the moment, I did not want to break actual backwards compatibility.

I'll remove the param and create a ticket for future improvement